examples/vucm/ni-daq/script.py

```diff
@@ -0,0 +1,93 @@
+import asyncio
+import csv
+import functools
+import io
+import os
+import socket
+
+import enapter
+
+def parse_bytes(bytes):
+    return bytes.decode("utf-8")
+
+
+def decode_csv(s):
+    csv_file = io.StringIO(s)
+    reader = csv.DictReader(csv_file)
+    return [row for row in reader]
+
+async def main():
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+        device_factory = functools.partial(
+            NIDAQ,
+            socket=sock,
+            tcp_port=os.environ["LISTEN_TCP_PORT"],
+        )
+        await enapter.vucm.run(device_factory)
+
+
+class NIDAQ(enapter.vucm.Device):
+    def __init__(self, socket, tcp_port, **kwargs):
+        super().__init__(**kwargs)
+        self.socket = socket
+        self.tcp_port = tcp_port
+
+    async def task_properties_sender(self):
+        while True:
+            await self.send_properties(
+                {
+                    "model": "NI cDAQ 9178",
+                }
+            )
+            await asyncio.sleep(10)
+
+    async def task_telemetry_sender(self):
+        server_address = ('localhost', int(self.tcp_port))
+        self.socket.bind(server_address)
+        self.socket.setblocking(False)
+        self.socket.listen(1)
+        
+        while True:
+            try:
+                await self.log.info('waiting for a connection')
+                connection, client_address = await asyncio.get_event_loop().sock_accept(self.socket)
+                
+                try:
+                    await self.log.info(f'connection from {client_address}')
+                    data_to_send = bytearray()
+                    
+                    while True:
+                        try:
+                            data = await asyncio.get_event_loop().sock_recv(connection, 1024)
+                            if not data:
+                                await self.log.info(f'no more data from {client_address}')
+                                break
+                            data_to_send.extend(data)
+                            await self.log.info(f'got data: {data}')
+                        except Exception as e:
+                            await self.log.error(f"Error receiving data: {e}")
+                            break
+                    
+                    await self.send_telemetry(
+                        {
+                            "status": parse_bytes(data_to_send),
+                        }
+                    )
+                    self.alerts.clear()
+                    
+                except Exception as e:
+                    self.alerts.add("nidaq_error")
+                    await self.log.error(f"failed to process data: {e}")
+                    
+                finally:
+                    connection.close()
+                    await asyncio.sleep(1)
+                    
+            except Exception as e:

```

Comment:
You if `break` up there, then there'll be no exception to catch down here. Do we really need the inner try-except block?